chain-abci/src/app/app_init.rs

```diff
@@ -244,6 +247,27 @@ fn store_valid_genesis_state(genesis_state: &ChainNodeState, inittx: &mut DBTran
     inittx.put(COL_EXTRA, b"init_chain_state", &encoded);
 }
 
+fn compute_accounts_root(account_storage: &mut AccountStorage, accounts: &[StakedState]) -> H256 {
+    let mut keys: Vec<_> = accounts.iter().map(StakedState::key).collect();
```

Comment:
let mut keys = accounts.iter().map(StakedState::key).collect();

Reply:
The `Vec<_>` is necessary here for `collect` to know the correct type.